homeassistant/components/tado/switch.py

```diff
@@ -0,0 +1,93 @@
+"""Module for Tado child lock switch entity."""
+
+import logging
+from typing import Any
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import TadoConfigEntry
+from .entity import TadoDataUpdateCoordinator, TadoZoneEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TadoChildLockSwitchEntity(TadoZoneEntity, SwitchEntity):
+    """Representation of a Tado child lock switch entity."""
+
+    def __init__(
+        self,
+        coordinator: TadoDataUpdateCoordinator,
+        zone_name: str,
+        zone_id: int,
+        device_info: dict[str, Any],
+    ) -> None:
+        """Initialize the Tado child lock switch entity."""
+        super().__init__(zone_name, coordinator.home_id, zone_id, coordinator)
+
+        self._device_info = device_info
+        self._device_id = self._device_info["shortSerialNo"]
+        self._attr_unique_id = f"{zone_id} {coordinator.home_id} child-lock"
+        self._attr_translation_key = "child_lock"
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn the entity on."""
+        await self.coordinator.set_child_lock(self._device_id, True)
+        await self.coordinator.async_request_refresh()
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn the entity off."""
+        await self.coordinator.set_child_lock(self._device_id, False)
+        await self.coordinator.async_request_refresh()
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._async_update_callback()
+        super()._handle_coordinator_update()
+
+    @callback
+    def _async_update_callback(self) -> None:
+        """Handle update callbacks."""
+        try:
+            self._device_info = self.coordinator.data["device"][self._device_id]
+        except KeyError:
+            _LOGGER.error(
+                "Could not update child lock info for device %s in zone %s",
+                self._device_id,
+                self.zone_name,
+            )
+        self._attr_is_on = self._device_info.get("childLockEnabled", False) is True
```

Comment:
```suggestion
        else:
            self._attr_is_on = self._device_info.get("childLockEnabled", False) is True
```
this should be on the else branch, right?

Reply:
I think it should update the state nevertheless.

Reply:
But is it ok to use `self._device_info` without getting a new instance? If so, do we need to always fetch a new instance?

Reply:
The fetching doesn't happen in this place. As far as I understand the data coordinator notifies subscribed entities if new data is available. 

I don't know if we really need to try/except but in case the device is deleted, I'm not sure if this deletion would reflect correctly for this entity. So better try to find the device info if available and nevertheless update the state from the cached (or updated) device info. 

It's really just some kind of "finally" and maybe it's better to put it in a "finally" block^^

Reply:
Done! And thanks for the heads up.